### PR TITLE
Remove RT build and replace with zero config CRA

### PR DESCRIPTION
**Pull request by bhavesh-desai-scratch**
Use CRA for Reactive Trader
Simplify NPM scripts

### CODE REVIEW

File: src/client/public/index.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (fdore) [inline src/client/public/index.html]: Reactive Trader Cloud

File: src/client/public/manifest.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Comment (fdore) [inline src/client/public/manifest.json]: should pbly change these

File: src/client/src/config/config.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+interface Config {
+  overwriteServerEndpoint: boolean,
+  serverEndPointUrl?: string,
```
Comment (fdore) [inline src/client/src/config/config.ts]: => serverEndpointUrl

File: src/client/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { parse } from 'query-string'
+import {run} from './notificationBootstrapper'
```
Comment (fdore) [inline src/client/src/index.tsx]: bracket spacing

File: src/client/src/ui/styles/scss/index.scss [hunk 5]
```diff
@@ -20,4 +20,5 @@
 @import 'popoutRegion.scss';
 @import 'spotTileControls.scss';
 @import 'variables';
+@import 'Alert-notification.scss';
```
Comment (fdore) [inline src/client/src/ui/styles/scss/index.scss]: alertNotification.scss for consistency

Comment (bhavesh-desai-scratch) [reply src/client/src/ui/styles/scss/index.scss]: For some strange reason, this broke the import! Will take a look again

File: src/client/src/system/openFin/openFin.ts [hunk 6]
```diff
@@ -202,13 +202,14 @@ export default class OpenFin {
...
+    if (!this.isRunningInOpenFin) { return }
 
     const tradeNotification = formatTradeNotification(trade, currencyPair)
+    // tslint:disable-next-line
     new fin.desktop.Notification({
-      url: '/notification.html',
+      url: '/index.html?notification=true',
       message: tradeNotification,
-      duration: 20000,
+      duration: 2000000,
```
[the remark this answers is not in the document]
Comment (bhavesh-desai-scratch) [reply src/client/src/system/openFin/openFin.ts]: Will fix. I did this to debug open-fin. Forgot to put it back.

File: src/client/src/services/referenceDataService.ts [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
-import { ServiceClient } from '../system/service'
 import '../system/observableExtensions/retryPolicyExt'
+import { ServiceClient } from '../system/service'
+import { ConnectionStatus, ServiceConst, UpdateType } from '../types'
+import { referenceDataMapper } from './mappers'
 
 const log = logger.create('ReferenceDataService')
 
-const getReferenceDataStream = (serviceClient, updateCache) => {
+const getReferenceDataStream = (serviceClient, updCache) => {
```
Comment (rikoe) [inline src/client/src/services/referenceDataService.ts]: I prefer whole words to abbreviations if we have code completion.

Comment (bhavesh-desai-scratch) [reply src/client/src/services/referenceDataService.ts]: Alot of these "fixes" are dure to shadowed variable names throughout the code base.

File: src/client/src/system/openFin/openFin.ts [hunk 8]
```diff
@@ -90,14 +90,14 @@ export default class OpenFin {
       if (this.limitCheckSubscriber === null) {
         log.debug('client side limit check not up, will delegate to to server')
         observer.next(true)
-        observer.compconste()
+        observer.complete()
```
[the remark this answers is not in the document]
Comment (rikoe) [reply src/client/src/system/openFin/openFin.ts]: 👍

File: src/client/src/system/service/serviceClient.ts [hunk 9]
```diff
@@ -156,8 +156,8 @@ export default class ServiceClient extends DisposableBase {
             )
           }
         },
-        err => o.error(err),
-        () => o.complete(),
+          err => o.error(err),
```
Comment (rikoe) [inline src/client/src/system/service/serviceClient.ts]: Indenting?

File: src/client/src/system/service/serviceClient.ts [hunk 10]
```diff
@@ -204,34 +205,34 @@ export default class ServiceClient extends DisposableBase {
                 () => {
                   o.complete()
                 },
-              ),
+            ),
```
Comment (rikoe) [inline src/client/src/system/service/serviceClient.ts]: Indenting?

File: src/client/src/ui/common/popout/openfinPopoutService.tsx [hunk 11]
```diff
@@ -1,9 +1,9 @@
+import * as _ from 'lodash'
 import * as React from 'react'
 import * as ReactDOM from 'react-dom'
 import OpenFinChrome from '../../shell/OpenFinChrome'
 import PopoutServiceBase from './popoutServiceBase'
-import * as _ from 'lodash'
-const DockingManager = require('exports-loader?DockingManager!../../../../lib/dockingManager.js')
+const {DockingManager} = require('./dockingManager.js')
```
Comment (rikoe) [inline src/client/src/ui/common/popout/openfinPopoutService.tsx]: Spacing

File: src/client/public/index.html (hunk 1 above)
Comment (bhavesh-desai-scratch) [reply src/client/public/index.html]: Done

File: src/client/tsconfig.json [hunk 12]
```diff
@@ -1,37 +1,34 @@
...
     "sourceMap": true,
-    "removeComments": true,
+    "allowJs": true,
     "jsx": "react",
+    "moduleResolution": "node",
+    "rootDir": "src",
     "forceConsistentCasingInFileNames": true,
-    "strict": true,
-    "declaration": false,
+    "noImplicitReturns": false,
```
Comment (rikoe) [inline src/client/tsconfig.json]: Why have we removed strict=true and now do individual settings? I thought the former was ideal, especially in terms of best practices for people looking at the code. Using strict and removing all the settings it covers can simplify the config a lot.

File: src/client/public/manifest.json (hunk 2 above)
Comment (bhavesh-desai-scratch) [reply src/client/public/manifest.json]: Done

File: src/client/src/config/config.ts (hunk 3 above)
Comment (bhavesh-desai-scratch) [reply src/client/src/config/config.ts]: Done

File: src/client/src/index.tsx (hunk 4 above)
Comment (bhavesh-desai-scratch) [reply src/client/src/index.tsx]: DONE. Also added auto formatting

File: src/client/src/main.tsx [hunk 13]
```diff
@@ -73,4 +74,9 @@ const appBootstrapper = () => {
 
 const runBootstrapper = location.pathname === '/' && location.hash.length === 0
 // if we're not the root we (perhaps a popup) we never re-run the bootstrap logic
-runBootstrapper && appBootstrapper()
+
+export function run(){
+    if(runBootstrapper){
+        appBootstrapper()
+      }
```
[the remark this answers is not in the document]
Comment (bhavesh-desai-scratch) [reply src/client/src/main.tsx]: Done and added hook

File: src/client/src/system/openFin/openFin.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (bhavesh-desai-scratch) [reply src/client/src/system/openFin/openFin.ts]: Done

File: src/client/src/system/service/autobahnConnectionProxyStub.ts [hunk 14]
```diff
@@ -94,27 +94,27 @@ class DummyPromise {
...
 
   get dto() {
-    return this._dto[0]
+    return this.dtoInner[0]
   }
 }
 
 class StubSubscribeResult extends DummyPromise {
-  _onResults
+  onResultsInner
```
[the remark this answers is not in the document]
Comment (bhavesh-desai-scratch) [reply src/client/src/system/service/autobahnConnectionProxyStub.ts]: Done

File: src/client/src/ui/styles/scss/index.scss (hunk 5 above)
Comment (bhavesh-desai-scratch) [reply src/client/src/ui/styles/scss/index.scss]: Done

File: src/client/webpack.config.js [hunk 15]
```diff
@@ -1,246 +0,0 @@
...
-
-    webpackConfig.devtool = 'source-map'
-
-    webpackConfig.plugins.push(
-      new webpack.HotModuleReplacementPlugin()
-    )
-  }
-
-  return webpackConfig
-}
```
Comment (rikoe) [inline src/client/webpack.config.js]: 👏

File: src/client/tsconfig.json (hunk 12 above)
Comment (bhavesh-desai-scratch) [reply src/client/tsconfig.json]: Fixed

**Pull request merged**